From 2e1803d3de80226c01d8fab2c03a63574cdcf603 Mon Sep 17 00:00:00 2001
Message-Id: <2e1803d3de80226c01d8fab2c03a63574cdcf603.1373271643.git.jdenemar@redhat.com>
From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Mon, 4 Mar 2013 20:50:42 +0000
Subject: [PATCH] Convert QEMU driver to use virLogProbablyLogMessage

https://bugzilla.redhat.com/show_bug.cgi?id=954248

The current QEMU code for skipping log messages only skips over
'debug' message, switch to virLogProbablyLogMessage to make sure
it skips over all of them
(cherry picked from commit 82793a2a5517c7f5f5f844aa03f0e0a019a04c2a)
---
 src/qemu/qemu_process.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/qemu/qemu_process.c b/src/qemu/qemu_process.c
index e9c45f0..908ca9b 100644
--- a/src/qemu/qemu_process.c
+++ b/src/qemu/qemu_process.c
@@ -1412,19 +1412,11 @@ qemuProcessReadLogOutput(virDomainObjPtr vm,
 {
     int retries = (timeout*10);
     int got = 0;
-    char *debug = NULL;
     int ret = -1;
     char *filter_next = buf;
 
     buf[0] = '\0';
 
-    /* This relies on log message format generated by virLogFormatString() and
-     * might need to be modified when message format changes. */
-    if (virAsprintf(&debug, ": %d: debug : ", vm->pid) < 0) {
-        virReportOOMError();
-        return -1;
-    }
-
     while (retries) {
         ssize_t func_ret, bytes;
         int isdead = 0;
@@ -1451,7 +1443,7 @@ qemuProcessReadLogOutput(virDomainObjPtr vm,
         /* Filter out debug messages from intermediate libvirt process */
         while ((eol = strchr(filter_next, '\n'))) {
             *eol = '\0';
-            if (strstr(filter_next, debug)) {
+            if (virLogProbablyLogMessage(filter_next)) {
                 memmove(filter_next, eol + 1, got - (eol - buf));
                 got -= eol + 1 - filter_next;
             } else {
@@ -1488,7 +1480,6 @@ qemuProcessReadLogOutput(virDomainObjPtr vm,
                    what, buf);
 
 cleanup:
-    VIR_FREE(debug);
     return ret;
 }
 
-- 
1.8.2.1

